diff --git a/srsepc/src/mme/nas.cc b/srsepc/src/mme/nas.cc
index 1633a2164..24bed1a79 100644
--- a/srsepc/src/mme/nas.cc
+++ b/srsepc/src/mme/nas.cc
@@ -708,6 +708,9 @@ bool nas::handle_service_request(uint32_t                m_tmsi,
 
     // Reset and store context with new mme s1ap id
     nas_ctx->reset();
+    // But don't erase imsi
+    emm_ctx->imsi  = imsi;
+    emm_ctx->state = EMM_STATE_DEREGISTERED;
     memcpy(&ecm_ctx->enb_sri, enb_sri, sizeof(struct sctp_sndrcvinfo));
     ecm_ctx->enb_ue_s1ap_id = enb_ue_s1ap_id;
     ecm_ctx->mme_ue_s1ap_id = s1ap->get_next_mme_ue_s1ap_id();
