diff --git a/srsenb/src/CMakeLists.txt b/srsenb/src/CMakeLists.txt
index f7aad9c91..7d8364746 100644
--- a/srsenb/src/CMakeLists.txt
+++ b/srsenb/src/CMakeLists.txt
@@ -36,7 +36,7 @@ endif (RPATH)
 add_library(enb_cfg_parser STATIC parser.cc enb_cfg_parser.cc)
 target_link_libraries(enb_cfg_parser ${LIBCONFIGPP_LIBRARIES})
 
-add_executable(srsenb main.cc enb.cc metrics_stdout.cc metrics_csv.cc metrics_json.cc metrics_ostatistic.cc)
+add_executable(srsenb-emane main.cc enb.cc metrics_stdout.cc metrics_csv.cc metrics_json.cc metrics_ostatistic.cc)
 
 set(SRSENB_SOURCES srsenb_phy srsenb_stack srsenb_common srsenb_upper srsenb_mac srsenb_rrc srslog system)
 set(SRSRAN_SOURCES srsran_common srsran_mac srsran_phy srsran_upper srsran_radio rrc_asn1 s1ap_asn1 enb_cfg_parser srslog system)
@@ -44,16 +44,16 @@ set(SRSRAN_SOURCES srsran_common srsran_mac srsran_phy srsran_upper srsran_radio
 set(SRSENB_SOURCES ${SRSENB_SOURCES} srsgnb_phy srsgnb_stack srsgnb_upper srsgnb_mac srsgnb_rrc)
 set(SRSRAN_SOURCES ${SRSRAN_SOURCES} rrc_nr_asn1 ngap_nr_asn1)
 
-target_link_libraries(srsenb  ${SRSENB_SOURCES}
-                              ${SRSRAN_SOURCES}
-                              ${CMAKE_THREAD_LIBS_INIT}
-                              ${Boost_LIBRARIES}
-                              ${SEC_LIBRARIES}
-                              ${LIBCONFIGPP_LIBRARIES}
-                              ${SCTP_LIBRARIES})
+target_link_libraries(srsenb-emane  ${SRSENB_SOURCES}
+                                    ${SRSRAN_SOURCES}
+                                    ${CMAKE_THREAD_LIBS_INIT}
+                                    ${Boost_LIBRARIES}
+                                    ${SEC_LIBRARIES}
+                                    ${LIBCONFIGPP_LIBRARIES}
+                                    ${SCTP_LIBRARIES})
 
 if (RPATH)
-  set_target_properties(srsenb PROPERTIES INSTALL_RPATH ".")
+  set_target_properties(srsenb-emane PROPERTIES INSTALL_RPATH ".")
 endif (RPATH)
 
 ########################################################################
@@ -61,9 +61,9 @@ endif (RPATH)
 ########################################################################
 if (NOT ${BUILDENB_CMD} STREQUAL "")
   message(STATUS "Added custom post-build-ENB command: ${BUILDENB_CMD}")
-  add_custom_command(TARGET srsenb POST_BUILD COMMAND ${BUILDENB_CMD})
+  add_custom_command(TARGET srsenb-emane POST_BUILD COMMAND ${BUILDENB_CMD})
 else(NOT ${BUILDENB_CMD} STREQUAL "")
   message(STATUS "No post-build-ENB command defined")
 endif (NOT ${BUILDENB_CMD} STREQUAL "")
 
-install(TARGETS srsenb DESTINATION ${RUNTIME_DIR})
+install(TARGETS srsenb-emane DESTINATION ${RUNTIME_DIR})
diff --git a/srsepc/src/CMakeLists.txt b/srsepc/src/CMakeLists.txt
index 7724dbc21..974e076a3 100644
--- a/srsepc/src/CMakeLists.txt
+++ b/srsepc/src/CMakeLists.txt
@@ -34,38 +34,38 @@ if (RPATH)
 endif (RPATH)
 
 
-add_executable(srsepc main.cc)
-target_link_libraries( srsepc  srsepc_mme
-                               srsepc_hss
-                               srsepc_sgw
-                               s1ap_asn1
-                               srsran_upper
-                               srsran_common
-                               srslog
-                               ${CMAKE_THREAD_LIBS_INIT}
-                               ${Boost_LIBRARIES}
-                               ${SEC_LIBRARIES}
-                               ${LIBCONFIGPP_LIBRARIES}
-                               ${SCTP_LIBRARIES}
-                               ${emanelte_LIBRARIES}
-                               ${ostatistic_LIBRARIES})
+add_executable(srsepc-emane main.cc)
+target_link_libraries(srsepc-emane  srsepc_mme
+                                    srsepc_hss
+                                    srsepc_sgw
+                                    s1ap_asn1
+                                    srsran_upper
+                                    srsran_common
+                                    srslog
+                                    ${CMAKE_THREAD_LIBS_INIT}
+                                    ${Boost_LIBRARIES}
+                                    ${SEC_LIBRARIES}
+                                    ${LIBCONFIGPP_LIBRARIES}
+                                    ${SCTP_LIBRARIES}
+                                    ${emanelte_LIBRARIES}
+                                    ${ostatistic_LIBRARIES})
 
-add_executable(srsmbms mbms-gw/main.cc )
-target_link_libraries(srsmbms   srsepc_mbms_gw
-                                srsran_upper
-                                srsran_common
-                                srslog
-                                ${CMAKE_THREAD_LIBS_INIT}
-                                ${Boost_LIBRARIES}
-                                ${SEC_LIBRARIES}
-                                ${LIBCONFIGPP_LIBRARIES}
-                                ${SCTP_LIBRARIES}
-                                ${emanelte_LIBRARIES}
-                                ${ostatistic_LIBRARIES})
+add_executable(srsmbms-emane mbms-gw/main.cc )
+target_link_libraries(srsmbms-emane   srsepc_mbms_gw
+                                      srsran_upper
+                                      srsran_common
+                                      srslog
+                                      ${CMAKE_THREAD_LIBS_INIT}
+                                      ${Boost_LIBRARIES}
+                                      ${SEC_LIBRARIES}
+                                      ${LIBCONFIGPP_LIBRARIES}
+                                      ${SCTP_LIBRARIES}
+                                      ${emanelte_LIBRARIES}
+                                      ${ostatistic_LIBRARIES})
 
 if (RPATH)
-  set_target_properties(srsepc PROPERTIES INSTALL_RPATH ".")
-  set_target_properties(srsmbms PROPERTIES INSTALL_RPATH ".")
+  set_target_properties(srsepc-emane PROPERTIES INSTALL_RPATH ".")
+  set_target_properties(srsmbms-emane PROPERTIES INSTALL_RPATH ".")
 endif (RPATH)
 
 ########################################################################
@@ -73,10 +73,10 @@ endif (RPATH)
 ########################################################################
 if (NOT ${BUILDEPC_CMD} STREQUAL "")
   message(STATUS "Added custom post-build-EPC command: ${BUILDENB_CMD}")
-  add_custom_command(TARGET srsenb POST_BUILD COMMAND ${BUILDENB_CMD})
+  add_custom_command(TARGET srsenb-emane POST_BUILD COMMAND ${BUILDENB_CMD})
 else(NOT ${BUILDEPC_CMD} STREQUAL "")
   message(STATUS "No post-build-EPC command defined")
 endif (NOT ${BUILDEPC_CMD} STREQUAL "")
 
-install(TARGETS srsepc DESTINATION ${RUNTIME_DIR})
-install(TARGETS srsmbms DESTINATION ${RUNTIME_DIR})
+install(TARGETS srsepc-emane DESTINATION ${RUNTIME_DIR})
+install(TARGETS srsmbms-emane DESTINATION ${RUNTIME_DIR})
diff --git a/srsue/src/CMakeLists.txt b/srsue/src/CMakeLists.txt
index a5ef34920..2b2557c6f 100644
--- a/srsue/src/CMakeLists.txt
+++ b/srsue/src/CMakeLists.txt
@@ -30,7 +30,7 @@ if (RPATH)
   set(CMAKE_BUILD_WITH_INSTALL_RPATH TRUE)
 endif (RPATH)
 
-add_executable(srsue main.cc ue.cc metrics_stdout.cc metrics_csv.cc metrics_ostatistic.cc)
+add_executable(srsue-emane main.cc ue.cc metrics_stdout.cc metrics_csv.cc metrics_ostatistic.cc)
 
 set(SRSUE_SOURCES srsue_phy srsue_stack srsue_upper srsue_mac srsue_rrc srslog system)
 set(SRSRAN_SOURCES srsran_common srsran_mac srsran_phy srsran_radio srsran_upper rrc_asn1 srslog system)
@@ -38,13 +38,13 @@ set(SRSRAN_SOURCES srsran_common srsran_mac srsran_phy srsran_radio srsran_upper
 set(SRSUE_SOURCES ${SRSUE_SOURCES} srsue_nr_stack srsue_rrc_nr srsue_mac_nr)
 set(SRSRAN_SOURCES ${SRSRAN_SOURCES} rrc_nr_asn1 ngap_nr_asn1)
 
-target_link_libraries(srsue   ${SRSUE_SOURCES}
-                              ${SRSRAN_SOURCES}
-                              ${CMAKE_THREAD_LIBS_INIT}
-                              ${Boost_LIBRARIES})
+target_link_libraries(srsue-emane   ${SRSUE_SOURCES}
+                                    ${SRSRAN_SOURCES}
+                                    ${CMAKE_THREAD_LIBS_INIT}
+                                    ${Boost_LIBRARIES})
 
 if (RPATH)
-  set_target_properties(srsue PROPERTIES INSTALL_RPATH ".")
+  set_target_properties(srsue-emane PROPERTIES INSTALL_RPATH ".")
 endif (RPATH)
 
 ########################################################################
@@ -69,9 +69,9 @@ endif(ZEROMQ_FOUND)
 ########################################################################
 if (NOT ${BUILDUE_CMD} STREQUAL "")
   message(STATUS "Added custom post-build-UE command: ${BUILDUE_CMD}")
-  add_custom_command(TARGET srsue POST_BUILD COMMAND ${BUILDUE_CMD})
+  add_custom_command(TARGET srsue-emane POST_BUILD COMMAND ${BUILDUE_CMD})
 else(NOT ${BUILDUE_CMD} STREQUAL "")
   message(STATUS "No post-build-UE command defined")
 endif (NOT ${BUILDUE_CMD} STREQUAL "")
 
-install(TARGETS srsue DESTINATION ${RUNTIME_DIR})
+install(TARGETS srsue-emane DESTINATION ${RUNTIME_DIR})
